api/src/main/scala/pl/touk/nussknacker/engine/api/process/ContextInitializer.scala

```diff
@@ -0,0 +1,77 @@
+package pl.touk.nussknacker.engine.api.process
+
+import cats.data.ValidatedNel
+import pl.touk.nussknacker.engine.api.context.ProcessCompilationError.NodeId
+import pl.touk.nussknacker.engine.api.context.{ProcessCompilationError, ValidationContext}
+import pl.touk.nussknacker.engine.api.runtimecontext.{ContextIdGenerator, EngineRuntimeContext}
+import pl.touk.nussknacker.engine.api.typed.typing.TypingResult
+import pl.touk.nussknacker.engine.api.{Context, Lifecycle, VariableConstants}
+
+/**
+  * ContextInitializer provides implementation of transformation from raw event generated by source to Context.
+  *
+  * @tparam Raw - type of raw event that is generated by source (KafkaDeserializationSchema for kafka) in source function.
+  */
+trait ContextInitializer[Raw] extends Serializable {
+
+  /**
+    * Initializes `Context` with raw event value.
+    *
+    * @param processId - id of the process, used to setup name of created Context
+    * @param nodeId  - name of node, used to setup name of created Context
+    */
+  def initContext(processId: String, nodeId: String): ContextInitializingFunction[Raw]
+
+  /**
+    * Enhances validation context with definition of all variables produced by the source.
+    *
+    * @param context      - `ValidationContext` initialized with global variables, definition of variables available in `Context` scope and their types
+    * @return - validation context with initialized "input" variable.
+    */
+  def validationContext(context: ValidationContext)(implicit nodeId: NodeId): ValidatedNel[ProcessCompilationError, ValidationContext]
+
+}
+
+trait ContextInitializingFunction[Raw] extends (Raw => Context) with Lifecycle with Serializable
+
+/**
+  * Basic implementation of context initializer. Used when raw event produced by source does not need further transformations and
+  * should be assigned to default "input" variable directly.
+  *
+  * @tparam Raw - type of raw event that is generated by source (KafkaDeserializationSchema for kafka) in source function.
+  */
+class BasicContextInitializer[Raw](protected val outputVariableType: TypingResult, protected val outputVariableName: String = VariableConstants.InputVariableName) extends ContextInitializer[Raw] {
+
+  // processId, nodeId is needed?
```

Comment:
is it TODO or redundant comment?

Reply:
Good catch - I forgot about it. It was TODO - processId is not needed because `ContextIdGenerator` is already aware of it so I removed it. NodeId is still needed because `EngineRuntimeContext` is scenario-wide not node-wide